core/src/impl/Kokkos_ZeroMemset_fwd.hpp

```diff
@@ -0,0 +1,29 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_ZEROMEMSET_FWD_HPP
+#define KOKKOS_ZEROMEMSET_FWD_HPP
+
+namespace Kokkos {
+namespace Impl {
+
+template <typename ExecutionSpace, class DT, class... DP>
```

Comment:
I was about to ask you to spell out what DT and DP... stand for but I am wondering now why it is not just templated on View instead.  Please check if it was intentional when ZeroMemset was introduced.

Reply:
The main reason was to follow the design of the other functions in `Kokkos_CopyViews.hpp` (`contiguous_fill`, `contiguous_fill_or_memset`, `deep_copy`, `local_deep_copy`, `local_deep_copy_contiguous`).

Reply:
I think that we should template on View and then we call always do
```C++
template <class DT, class... DP>
struct ZeroMemset<my-execution-space, View<DT, DP...>> { /* ... */ };
```

and maybe spell out what "DT" ("data type"?? "destination type") and "DP" are

Reply:
We oftentimes use `View<T, P...>` but we don't really need to know about the template parameters here anyway. I would also be happy with a comment.

Reply:
Would you mind trying
```C++
template <typename ExecutionSpace, typename View>
struct ZeroMemset;

template <typename T, typename... P>
struct ZeroMemset<Cuda, Kokkos::View<T, P...>>
{};
```
(or we do it in a follow-up).